From 062ef52c8a417aa90751e30167d02b4b5a0c0850 Mon Sep 17 00:00:00 2001
From: Jakub Jelen <jjelen@redhat.com>
Date: Tue, 27 Oct 2026 12:09:01 +0100
Subject: [PATCH] Verify source signatures with openpgpverify

Replace the GnuPG-based %gpgverify with %openpgpverify, which uses
Sequoia sqv and takes the same options, as part of the Fedora 45 Change
https://fedoraproject.org/wiki/Changes/Sequoia_openpgpverify

Refresh nettle-release-keyring.gpg from keyservers/WKD: same
certificate(s), with self-signatures our copy was missing; the commands
are in the spec comment.
Release bumped with a changelog entry.

Assisted-by: Claude Opus 5.5
---
 mingw-nettle.spec | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/mingw-nettle.spec b/mingw-nettle.spec
index 98a9de8..0dda9ad 100644
--- a/mingw-nettle.spec
+++ b/mingw-nettle.spec
@@ -2,7 +2,7 @@
 
 Name:           mingw-nettle
 Version:        3.10.2
-Release:        3%{?dist}
+Release:        4%{?dist}
 
 Summary: MinGW package for nettle cryptographic library
 # Automatically converted from old format: LGPLv3+ or GPLv2+ - review is highly recommended.
@@ -11,6 +11,10 @@ URL:    http://www.lysator.liu.se/~nisse/nettle/
 
 Source0: http://www.lysator.liu.se/~nisse/archive/nettle-%{version}.tar.gz
 Source1: http://www.lysator.liu.se/~nisse/archive/nettle-%{version}.tar.gz.sig
+# Refreshed from keyservers/WKD (same keys, newer signatures):
+#   sq --home=none network search --output - 343C2FF0FBEE5EC2EDBEF399F3599FF828C67298 | sq keyring filter --experimental --cert 343C2FF0FBEE5EC2EDBEF399F3599FF828C67298 --output update.pgp
+#   sq keyring merge --output nettle-release-keyring.gpg.new nettle-release-keyring.gpg update.pgp
+#   mv nettle-release-keyring.gpg.new nettle-release-keyring.gpg
 Source2: nettle-release-keyring.gpg
 # MinGW does not support explicit_bzero()
 #Patch0:  nettle-3.8-zeroize-stack.patch
@@ -33,6 +37,7 @@ BuildRequires:  mingw64-openssl
 BuildRequires:  autoconf, automake
 BuildRequires:  gcc
 BuildRequires:  m4
+BuildRequires: openpgpverify
 
 
 %description
@@ -71,7 +76,7 @@ or even in kernel space.
 
 %prep
 %autosetup -Tb 0 -p1 -n nettle-%{version}
-%{gpgverify} --keyring='%{SOURCE2}' --signature='%{SOURCE1}' --data='%{SOURCE0}'
+%{openpgpverify} --keyring='%{SOURCE2}' --signature='%{SOURCE1}' --data='%{SOURCE0}'
 
 
 %build
@@ -132,6 +137,9 @@ rm -rf $RPM_BUILD_ROOT%{mingw64_infodir}/
 
 
 %changelog
+* Fri Oct 23 2026 Jakub Jelen <jjelen@redhat.com> - 3.10.2-4
+- Verify upstream source signatures with openpgpverify
+
 * Thu Jul 16 2026 Fedora Release Engineering <releng@fedoraproject.org> - 3.10.2-3
 - Rebuilt for https://fedoraproject.org/wiki/Fedora_45_Mass_Rebuild
 
-- 
2.56.0

